### PR TITLE
Trit & H2 Consistency Fix

### PR DESCRIPTION
**Pull request by vincentiusvin**
## About The Pull Request
Tritiumfire has used the wrong variable for around three years. This fixes that.
Also some code improvements to the burned_fuel variable.

## Why It's Good For The Game
Consistency is good.

## Changelog
:cl:
fix: tritium and h2 now uses burned_fuel for o2 consumption.
code: moved the burned_fuel amount around so its tidier.
/:cl:

### CODE REVIEW

File: code/modules/atmospherics/gasmixtures/reactions.dm [hunk 1]
```diff
@@ -160,22 +160,27 @@ nobiliumsuppression = INFINITY
 	cached_results["fire"] = 0
 	var/turf/open/location = isturf(holder) ? holder : null
 	var/burned_fuel = 0
+
 	if(cached_gases[/datum/gas/oxygen][MOLES] < cached_gases[/datum/gas/tritium][MOLES] || MINIMUM_TRIT_OXYBURN_ENERGY > air.thermal_energy())
 		burned_fuel = cached_gases[/datum/gas/oxygen][MOLES] / TRITIUM_BURN_OXY_FACTOR
 		cached_gases[/datum/gas/tritium][MOLES] -= burned_fuel
+
+		ASSERT_GAS(/datum/gas/water_vapor, air) //oxygen+more-or-less hydrogen=H2O
+		cached_gases[/datum/gas/water_vapor][MOLES] += burned_fuel / TRITIUM_BURN_OXY_FACTOR
```
Comment (LemonInTheDark) [inline code/modules/atmospherics/gasmixtures/reactions.dm:169]: Why move this up?

File: code/modules/atmospherics/gasmixtures/reactions.dm [hunk 2]
```diff
@@ -160,22 +160,27 @@ nobiliumsuppression = INFINITY
...
+		
 	else
-		burned_fuel = cached_gases[/datum/gas/tritium][MOLES] * TRITIUM_BURN_TRIT_FACTOR
-		cached_gases[/datum/gas/tritium][MOLES] -= cached_gases[/datum/gas/tritium][MOLES] / TRITIUM_BURN_TRIT_FACTOR
-		cached_gases[/datum/gas/oxygen][MOLES] -= cached_gases[/datum/gas/tritium][MOLES]
+		burned_fuel = cached_gases[/datum/gas/tritium][MOLES]
+
+		cached_gases[/datum/gas/tritium][MOLES] -= burned_fuel / TRITIUM_BURN_TRIT_FACTOR
+		cached_gases[/datum/gas/oxygen][MOLES] -= burned_fuel
```
Comment (LemonInTheDark) [inline code/modules/atmospherics/gasmixtures/reactions.dm:175]: Could you elaborate on why you thing `TRITIUM_BURN_TRIT_FACTOR` was used improperly here before?

File: code/__DEFINES/reactions.dm [hunk 3]
```diff
@@ -7,7 +7,7 @@
 #define PLASMA_MINIMUM_OXYGEN_NEEDED		2
 #define PLASMA_MINIMUM_OXYGEN_PLASMA_RATIO	30
 #define FIRE_CARBON_ENERGY_RELEASED			100000	//Amount of heat released per mole of burnt carbon into the tile
-#define FIRE_HYDROGEN_ENERGY_RELEASED		280000  //Amount of heat released per mole of burnt hydrogen and/or tritium(hydrogen isotope)
+#define FIRE_HYDROGEN_ENERGY_RELEASED		2800000  //Amount of heat released per mole of burnt hydrogen and/or tritium(hydrogen isotope)
```
Comment (LemonInTheDark) [inline code/__DEFINES/reactions.dm:10]: Was this change intentional?

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]

File: code/__DEFINES/reactions.dm (hunk 3 above)
Comment (vincentiusvin) [reply code/__DEFINES/reactions.dm:10]: yes, i removed the *10 from the burned_fuel part of the code and moved it there to make it official.

File: code/modules/atmospherics/gasmixtures/reactions.dm (hunk 1 above)
Comment (vincentiusvin) [reply code/modules/atmospherics/gasmixtures/reactions.dm:169]: Since burned_fuel is multiplied by 10 for saturated fires, it produces more water per reaction tick for saturated fires than unsaturated fires. This accounts for that

File: code/modules/atmospherics/gasmixtures/reactions.dm (hunk 2 above)
Comment (vincentiusvin) [reply code/modules/atmospherics/gasmixtures/reactions.dm:175]: it doesnt make sense that burned_fuel is multiplied by 10. I just moved it around for the sake of sanity and to better reflect the (supposed) meaning of the variable

File: code/modules/atmospherics/gasmixtures/reactions.dm (hunk 1 above)
Comment (LemonInTheDark) [reply code/modules/atmospherics/gasmixtures/reactions.dm:169]: but burned_fuel is only modified once, cause of the else. Isn't keeping it in the `if(burned_fuel)` block the same behavior? or am I missing something.

File: code/modules/atmospherics/gasmixtures/reactions.dm (hunk 2 above)
Comment (LemonInTheDark) [reply code/modules/atmospherics/gasmixtures/reactions.dm:175]: By the looks of things burned_fuel is used for h2o/rad output and dealing with hotspots, whereas the the burning of trit/o2 is based purely on the amount of trit.

Is your goal to change the behavior of the reaction? or to clean up the code. Cause what you're doing not for sure changes behavior, not least of which because trit fires will burn out more easily in this special case.

File: code/modules/atmospherics/gasmixtures/reactions.dm (hunk 1 above)
Comment (vincentiusvin) [reply code/modules/atmospherics/gasmixtures/reactions.dm:169]: the old if(burned_fuel) part divides burned_fuel by 100 and saturated burns multiplies it by 10. Since i removed the *10 for burned_fuel, keeping it there would make saturated fires produce 10 times less water vapor. By moving it into the if-else part, it keeps the old behavior of producing 10 times more water vapour for saturated fires per reaction tick.

File: code/modules/atmospherics/gasmixtures/reactions.dm (hunk 2 above)
Comment (vincentiusvin) [reply code/modules/atmospherics/gasmixtures/reactions.dm:175]: I initially wanted to just make a surgical change to the o2 consumption rate and also fix the burned_fuel snowflake bit, but i forgot to account for the rad output etc. It seems like this will indeed make it quite the behavior change. I'll try and account for everything and keep the behavior on the next commits.

**Comment by vincentiusvin** [comment]
running numbers through, this is the result that i get, with old code compared to the new code.
image

File: code/modules/atmospherics/gasmixtures/reactions.dm (hunk 1 above)
Comment (LemonInTheDark) [reply code/modules/atmospherics/gasmixtures/reactions.dm:169]: Ah I see

**Comment by LemonInTheDark** [comment]
If you're going to change behavior I'd like an explanation of why it's better/more interesting, I'd prefer the first option in theory but I'm somewhat concerned that it would bloat the code a bit too much.

Do you mind elaborating on what you dislike about the current state of things so I can better understand your position? is it purely a readability thing? or is it about consistency with other reactions.

**Comment by vincentiusvin** [comment]
> Do you mind elaborating on what you dislike about the current state of things so I can better understand your position? is it purely a readability thing? or is it about consistency with other reactions.

I dislike how it isn't consistent with other reactions.

**Comment by vincentiusvin** [comment]
Now it should be fully similar to the old one. All except in the cached_results bit used for fire calculations.

**Comment by LemonInTheDark** [comment]
At a glance it looks like the output cached_results value has changed, I'd like that to stay the same if possible, unless there's a reason to change it
